pinot-core/src/main/java/org/apache/pinot/core/operator/filter/PrioritizedFilterOperator.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.filter;
+
+import java.util.OptionalInt;
+import org.apache.pinot.core.common.Block;
+import org.apache.pinot.core.common.Operator;
+
+
+/**
+ * Operators that implements this interface can define how to be sorted in an AND, as defined in
+ * {@link FilterOperatorUtils.Implementation#}.
+ */
+public interface PrioritizedFilterOperator<T extends Block> extends Operator<T> {
+
+  int HIGH_PRIORITY = 0;
```

Comment:
Bucketizing in High/Medium/Low might be too coarse grain? Can we have multiple within a bucket? If so, how would the relative ordering be defined?

Reply:
> Bucketizing in High/Medium/Low might be too coarse grain?

Totally, but these are the values we are currently using. I initially multiplied the numbers by 100, but finally decided to keep the same values in order to keep this PR as simple as possible. I guess we can do the change here given that is not just me the one that though about that. The number is only used in an anonymous comparator, so I don't imagine it may have an actual impact.

>  If so, how would the relative ordering be defined?

The number is only used in a comparator, so if two operators have the same weight, it means they the order between them is not going to be always the same.